Modules/_collectionsmodule.c

```diff
@@ -537,12 +538,22 @@ deque_copy(PyObject *deque)
         Py_DECREF(new_deque);
         return NULL;
     }
-    if (old_deque->maxlen < 0)
-        return PyObject_CallFunctionObjArgs((PyObject *)(Py_TYPE(deque)),
-                                            deque, NULL);
-    else
-        return PyObject_CallFunction((PyObject *)(Py_TYPE(deque)), "Oi",
-            deque, old_deque->maxlen, NULL);
+    if (old_deque->maxlen < 0) {
+        result = PyObject_CallFunctionObjArgs((PyObject *)(Py_TYPE(deque)),
+                                              deque, NULL);
+    }
+    else {
+        result = PyObject_CallFunction((PyObject *)(Py_TYPE(deque)), "Oi",
+                                       deque, old_deque->maxlen, NULL);
+    }
```

Comment:
Please restore 541-548 to their former state and limit the patch to just the issue at hand.